internal/subshell/subshell.go

```diff
@@ -239,24 +244,21 @@ func DetectShell(cfg sscommon.Configurable) (string, string) {
 	return name, path
 }
 
-func detectShellWindows() string {
-	// Windows does not provide a way of identifying which shell we are running in, so we have to look at the parent
-	// process.
-
+func detectShellParent() string {
 	p, err := process.NewProcess(int32(os.Getppid()))
 	if err != nil && !errors.As(err, ptr.To(&os.PathError{})) {
-		panic(err)
+		logging.Error("Failed to get parent process: %v", err)
```

Comment:
Not sure why this was a panic, but we went from panicking to essentially ignoring it. Please return the error up the chain instead.

Reply:
This was originally a windows only function which may be a reason for the panic?

No other functions in this package return errors either as `DetectShell` has fallbacks so it always returns a shell name. This function is also used but `subshell.New`. So even if we get a fallback shell from either the `SHELL` env var or the system default we would error out because we couldn't detect the parent.

Again, I just want to check before making this change.